code/game/turfs/simulated/floor/plating.dm

```diff
@@ -173,4 +173,59 @@
 
 /turf/simulated/floor/plating/abductor/New()
 	..()
-	icon_state = "alienpod[rand(1,9)]"
\ No newline at end of file
+	icon_state = "alienpod[rand(1,9)]"
+
+/turf/simulated/floor/plating/basalt
+	name = "basalt floor"
+	icon_state = "basalt1"
+
+/turf/simulated/floor/plating/basalt/New()
+	..()
+	icon_state = "basalt[rand(1,12)]"
+
+/turf/simulated/floor/plating/hole
+	name = "hole"
+	desc = "Yep, thats a hole."
+	icon = 'icons/turf/floors/hole.dmi'
+	icon_state = "hole"
+	smooth = SMOOTH_TRUE
+	canSmoothWith = null
+	mouse_opacity = 0
+	var/tp_x = 0 //hole drop-point
+	var/tp_y = 0
+	var/tp_z = 0
+
+/turf/simulated/floor/plating/hole/New()
+	..()
+	spawn(1)
+		update_icon()
+
+/turf/simulated/floor/plating/hole/update_icon()
+	if(!..())
+		return 0
+	if(!broken && !burnt)
+		if(smooth)
+			smooth_icon(src)
+	else
+		make_plating()
+		if(smooth)
+			smooth_icon_neighbors(src)
+
+/turf/simulated/floor/plating/hole/Entered(atom/movable/H)
+	if(!istype(H))
+		return
+	if(istype(H, /obj))
+		var/obj/O = H
+		O.visible_message("[H] falls down \the [src]!")
+		O.x = tp_x
+		O.y = tp_y
+		O.z = tp_z
+	else if (istype(H, /mob/living))
+		var/mob/living/M = H
+		M.visible_message("[M] falls down \the [src]!", \
+			"<span class='warning'>You fall down \the [src]!</span>")
+		M.x = tp_x
+		M.y = tp_y
+		M.z = tp_z
+		M.Weaken(6)
+		M.emote("scream")
```

Comment:
i had this part looked over in coderbus by 3 people in a piratepad and none of them said anything bad about it. im just a sub-par coder trying to make shit for my map fuck